docs: add documentation for user group syncing (#2314)
Unleash/unleash: website/docs/how-to/how-to-set-up-group-sso-sync.md

---
title: How set up user group SSO syncing
---

:::info availability

> User group syncing is planned to be released in Unleash 4.18 and will be available for enterprise customers.

Reviewer: Same here as with the previous availability note. We may want to say that it's at "some point in the future".

:::

This guide takes you through how to configure your [user groups](../user_guide/rbac#user-groups) to automatically populate users through Single Sign On (SSO). Refer to [setting up Keycloak for user group sync](../../advanced/sso-keycloak-group-sync) for an end to end example. Note that the steps below require you to be logged in as an admin user.

> ### Step 1: Navigate to SSO configuration {#step-1}

Reviewer: These should be one level higher

```suggestion
## Step 1: Navigate to SSO configuration {#step-1}
```

Navigate to the "Single sign-on" configuration page.

![The Unleash Admin UI with the steps highlighted to navigate to the Single sign-on configuration.](/img/setup-sso-group-sync-1.png)

> ### Step 2: Enable Group Syncing {#step-2}

Reviewer: ```suggestion
## Step 2: Enable Group Syncing {#step-2}
```

> Turn on "Enable Group Syncing" and input a field a for "Group Field JSON Path". The latter is the JSON path in the token response where your group properties are located, this is up to your SSO provider, a full example for Keycloak can be [found here](../../advanced/sso-keycloak-group-sync). Once you're happy, save your configuration.

Reviewer: Some small rewording suggestions for clarity here. This is a tricky paragraph. I think the behavior of the "Group Field JSON Path" should be explained in the reference docs (rbac).

```suggestion
Turn on "Enable Group Syncing" and enter a value a for "Group Field JSON Path". Refer to the [User group SSO integration documentation](../user_guide/rbac.md#user-group-sso-integration) for more information or to the [how-to guide for integrating with Keycloak](how-to-set-up-keycloak-sso.md) for a practical example. 

The value is the JSON path in the token response where your group properties are located, this is up to your SSO provider, a full example for Keycloak can be [found here](../../advanced/sso-keycloak-group-sync). Once you're happy, save your configuration.
```

![The Single sign-on configuration page with enable group syncing, group field JSON path and save inputs highlighted.](/img/setup-sso-group-sync-2.png)

> ### Step 3: Navigate to a group {#step-3}

Reviewer: ```suggestion
## Step 3: Navigate to a group {#step-3}
```

Navigate to the group you want to sync users for.

![The Unleash Admin UI with the steps highlighted to navigate to groups and a highlighted group card.](/img/setup-sso-group-sync-3.png)

> ### Step 4: Edit the group configuration {#step-4}

Reviewer: ```suggestion
## Step 4: Edit the group configuration {#step-4}
```

Navigate to edit group.

![The group configuration screen with edit group highlighted.](/img/setup-sso-group-sync-4.png)

> ### Step 5: Link SSO groups to your group {#step-5}

Reviewer: ```suggestion
## Step 5: Link SSO groups to your group {#step-5}
```
